### PR TITLE
DDCYLS-3867 | Add Confirm Company Details page and corresponding kick-out page

### PR DESCRIPTION
**Pull request by jpatton638**

### CODE REVIEW

File: test/models/AddressSpec.scala [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+             |  "countryCode":"$countryCode"
+             |}
+             |""".stripMargin
+
+        an[IllegalArgumentException] mustBe thrownBy {
+          Json.parse(addressJsonStr).as[Address]
+        }
+      }
+
+      "when both postcode and country are both missing" in {
```
Comment (sunmingdh) [inline test/models/AddressSpec.scala:114]:
```suggestion
      "when both postcode and country are missing" in {
```

File: app/models/Address.scala [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  private def applyJson(line1: String,
+                        line2: Option[String],
+                        line3: Option[String],
+                        line4: Option[String],
+                        postalCode: Option[String],
+                        countryCode: Option[String]): Address = {
+    (postalCode, countryCode) match {
+      case (Some(postcode), None) => UkAddress(line1, line2, line3, line4, postcode)
+      case (None, Some(countryCode)) => InternationalAddress(line1, line2, line3, line4, countryCode)
+      case (Some(postcode), Some(countryCode)) if countryCode == "GB" => UkAddress(line1, line2, line3, line4, postcode)
```
Comment (sunmingdh) [inline app/models/Address.scala:53]:
```suggestion
      case (Some(postcode), Some("GB")) => UkAddress(line1, line2, line3, line4, postcode)
```

File: app/config/FrontendAppConfig.scala [hunk 3]
```diff
@@ -30,6 +30,7 @@ class FrontendAppConfig @Inject() (configuration: Configuration, servicesConfig:
 
   private val contactHost                  = configuration.get[String]("contact-frontend.host")
   private val contactFormServiceIdentifier = "digital-service-tax-registration-frontend"
+  val dstBackendBaseUrl: String            = servicesConfig.baseUrl("digital-services-tax")
```
Comment (sunmingdh) [inline app/config/FrontendAppConfig.scala:33]: do we need to add `dstBackendBaseUrl` to other Envs, such as `app-config-base`?

File: conf/application.conf [hunk 4]
```diff
@@ -37,6 +37,11 @@ microservice {
         host     = localhost
         port     = 8500
       }
+      digital-services-tax {
+        protocol = http
+        host     = localhost
+        port     = 8741
+      }
```
Comment (sunmingdh) [inline conf/application.conf:44]: do we need to add `digital-services-tax` to other Envs, such as `app-config-base`?

Comment (jpatton638) [reply conf/application.conf:44]: I'll raise a PR for this, thanks!

Comment (jpatton638) [reply conf/application.conf:44]: https://github.com/hmrc/app-config-base/pull/11290

**Review by sunmingdh — APPROVED** [review]

**Pull request merged**